Blocks/Persistence/file.h

```diff
@@ -192,8 +199,16 @@ class FilePersister {
                 }
                 head = us;
                 head_offset = std::streamoff(current_offset) + offset;
-              } else {
-                head_offset = 0;
+              } else if (!value.compare(0, signature_key_length, constants::kSignatureDirective)) {
+                // The signature could be only at the beginning of the file.
+                if (current_offset != 0) {
+                  CURRENT_THROW(InvalidSignaturePosition());
+                }
+                auto offset = signature_key_length;
+                while (offset < value.length() && std::isspace(value[offset])) ++offset;
+                if (value.compare(offset, signature.length(), signature)) {
+                  CURRENT_THROW(InvalidStreamSignature());
```

Comment:
A diff between old and new signatures would be nice here. I mean, just print them; the user would be able to run `cut` and `jq` on their own.